tfx/components/trainer/executor.py

```diff
@@ -166,6 +192,46 @@ def _GetFnArgs(self, input_dict: Dict[Text, List[types.Artifact]],
         # Additional parameters to pass to trainer function.
         **custom_config)
 
+
+  def _CleanServingDir(self, working_dir: Text) -> None:
+    """Copies exported serving model to base directory and deletes
+    non-serving model files from serving model directory.
+
+    Checks model artifact directory structure to determine a Keras or
+    Estimator model was exported. If Estimator model, exported
+    serving model is copied to base serving model directory, and all
+    other files are deleted.
+    """
+    model_export_dir = _serving_model_path(working_dir)
+    serving_model_dir = path_utils.serving_model_dir(working_dir)
+
+    if model_export_dir != serving_model_dir:
+      # Copy serving model export to base serving model directory
+      tempdir = tempfile.mkdtemp()
```

Comment:
use self.working_dir(), it will create a temp path under pipeline_root.

I'm not sure if temfile works on Cloud or other file systems